app.js

```diff
@@ -28,9 +31,65 @@ app.use(express.urlencoded({ extended: true }));
 app.use(session(sess));
 
 app.use(express.json());
-app.use("/api/v1/query-info", v1QueryInfo);
+app.get("/api/v1/engines", async (req, res) => {
+    const path = "engines"
+    const folders = fs.readdirSync(path).filter(f => fs.lstatSync(`${path}/${f}`).isDirectory());
+
+    res.send({
+        success: true, 
+        engines: folders 
+    });
+    return;
+})
+
+app.get("/api/v1/engines/:engine/parameters", async (req, res) => {
```

Comment:
I prefer to put each API call in its own file named by its route.  This helps me easily browse the API from a code perspective